extensions/wikia/AdEngine/js/SlotTweaker.js

```diff
@@ -179,6 +182,47 @@ define('ext.wikia.adEngine.slotTweaker', [
 		return;
 	}
 
+	function collapse(slotName) {
+		var slot = doc.getElementById(slotName);
+
+		slot.style.maxHeight = slot.scrollHeight + 'px';
+		noop(slot.offsetHeight);
+		DOMElementTweaker.addClass(slot, 'slot-animation');
+		slot.style.maxHeight = '0';
+	}
+
+	function expand(slotName) {
+		var slot = doc.getElementById(slotName);
+
+		slot.style.maxHeight = slot.offsetHeight + 'px';
+		DOMElementTweaker.removeClass(slot, 'hidden');
+		DOMElementTweaker.addClass(slot, 'slot-animation');
+		slot.style.maxHeight = slot.scrollHeight + 'px';
+	}
+
+	function messageCallback(data) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Function has a complexity of 13 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)